Backend/app/kafka-consumers/src/Consumer/KafkaTable/Processor.hs

```diff
@@ -0,0 +1,52 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Consumer.KafkaTable.Processor
+  ( kafkaTableProcessor,
+  )
+where
+
+import qualified AWS.S3 as S3
+import qualified Data.Aeson as A
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import Environment
+import Kernel.Prelude
+import Kernel.Streaming.Kafka.KafkaTable as Kafka
+import Kernel.Types.Error
+import Kernel.Utils.Common (decodeFromText, encodeToText, logDebug, logWarning, throwError)
+
+-- for now kafkaTable.tableName = "beckn_request", in future can be added other tables
+kafkaTableProcessor :: Map.Map String [Kafka.KafkaTable] -> Flow ()
+kafkaTableProcessor mapKafkaTable = do
+  pathPrefix <- T.unpack <$> asks (.s3Env.pathPrefix)
+  void $
+    flip Map.traverseWithKey mapKafkaTable $ \filePathWithoutPrefix mappedKafkaTables -> do
+      let filePath = pathPrefix <> "/" <> filePathWithoutPrefix
+      logDebug $ "MessagesReceived: " <> show (length mappedKafkaTables) <> "; filePath: " <> show filePath
+      existingFile <- handle @Flow @SomeException (pure . const "") (S3.get filePath)
+      if T.null existingFile
+        then do
+          logDebug $ "Create new file: " <> show filePath
+          S3.put filePath (encodeToText @[A.Value] (mappedKafkaTables <&> (.tableContent))) -- normal case
+        else do
+          let mbExistingRequests :: Maybe [A.Value] = decodeFromText existingFile
```

Comment:
do we need to decode the existing file ? can this be avoided ?

Reply:
We decode only when we have some drainer delay. Say, cronJob started at 12:30. Then file `atlas_app/beckn_requests/2023.10.25-11.json` does not exist yet and we don't need to decode anything. If we have some outdated value from `atlas_app/beckn_requests/2023.10.25-10.json` then we need to decode and append this value.

I can append each entry as separate line, as you said in other comment:

```
{"timestamp": "2023-09-25T16:00:18.196509901Z", "schemaName" : "atlas_app", "tableName" : "beckn_request", "tableContent" : {<JSON_TABLE_CONTENT_OBJECT_HERE>}}
{"timestamp": "2023-09-25T16:00:19.196509901Z", "schemaName" : "atlas_driver_offer_bpp", "tableName" : "beckn_request", "tableContent" : {<JSON_TABLE_CONTENT_OBJECT_HERE>}}
```
with this approach we don't need encode/decode list of objects, each object will be decoded as separate line in file

Reply:
fixed, now each entry will append as separate line, and no decode/encode required, just append lines. Example for `estimate_breakup` table:

```
{"priceCurrency":"INR","estimateId":"d94c3195-3733-4ee2-9d38-2c0b127cfbb3","id":"fd8e2f7e-5ce8-4109-afac-7e85f7f4b886","title":"DRIVER_MIN_EXTRA_FEE","priceValue":10}
{"priceCurrency":"INR","estimateId":"d94c3195-3733-4ee2-9d38-2c0b127cfbb3","id":"291a3ed4-cbfc-497a-bb99-806f8fcd15f5","title":"EXTRA_PER_KM_FARE","priceValue":12}
{"priceCurrency":"INR","estimateId":"d94c3195-3733-4ee2-9d38-2c0b127cfbb3","id":"d7a905e4-6bba-43d4-9afa-047957129eb2","title":"BASE_DISTANCE_FARE","priceValue":10}
```